tools/dgeni/processors/categorizer.js

```diff
@@ -171,6 +179,18 @@ function getDirectiveOutputAlias(doc) {
   return isDirectiveOutput(doc) ? doc.decorators.find(d => d.name == 'Output').arguments[0] : '';
 }
 
+function getDirectiveSelectors(classDoc) {
+  let metadata = classDoc.decorators
+    .find(d => d.name === 'Component' || d.name === 'Directive').arguments[0];
+
+  let selectorMatches = /selector\s*:\s*(?:"|')([^']*?)(?:"|')/g.exec(metadata);
+  selectorMatches = selectorMatches && selectorMatches[1];
+
+  return selectorMatches ? selectorMatches.split(/\s*,\s*/)
+    .filter(match => match !== '' && !match.includes('mat') && !DIRECTIVE_SELECTOR_BLACKLIST.includes(match))
+    : selectorMatches;
```

Comment:
I'd shorten `match` to just `s` so that it's
```ts
return selectorMatches ? 
    selectorMatches.split(/\s*,\s*/)
        .filter(s => s && !s.includes('mat') && !SELECTOR_BLACKLIST.has(s))
   : selectorMatches;
```